### PR TITLE
ARROW-7369: [GLib] Add garrow_table_combine_chunks

### PR DESCRIPTION
**Pull request by mrkn**

### CODE REVIEW

File: c_glib/test/test-table.rb [hunk 1]
```diff
@@ -210,5 +210,21 @@ def test_concatenate
...
+                                               build_boolean_array([false])])
+      )
+      combined_table = table.combine_chunks
+      assert_equal(table, combined_table)
+
+      chunked_array = combined_table.get_column_data(0)
+      assert_equal(1, chunked_array.n_chunks)
+
+      assert_equal([true, false, true, false, true, false],
+                   chunked_array.get_chunk(0).values)
```
Comment (kou) [inline c_glib/test/test-table.rb]: One assertion is better than multiple assertions for debug propose:

```ruby
combined_table = table.combine_chunks
all_values = combined_table.n_chunks.times.collect do |i|
  combined_table.get_column_data(i).values
end
assert_equal([[true, false, true, false, true, false]],
             all_values)
```

if `assert_equal(1, chunked_array.n_chunks)` is failed, we only know the number of wrong chunks. But the above one assertion shows us not only the number of chunks but also details of each chunk.

File: c_glib/arrow-glib/table.cpp [hunk 2]
```diff
@@ -588,6 +588,32 @@ garrow_table_slice(GArrowTable *table,
...
+GArrowTable *
+garrow_table_combine_chunks(GArrowTable *table,
+                            GError **error)
+{
+  const auto arrow_table = garrow_table_get_raw(table);
+
+  std::shared_ptr<arrow::Table> arrow_combined_table;
+  auto status = arrow_table->CombineChunks(arrow::default_memory_pool(),
+                                           &arrow_combined_table);
+  if (garrow_error_check(error, status, "[table][combine_chunks]")) {
```
Comment (kou) [inline c_glib/arrow-glib/table.cpp]: Could you use `-` for word separator? `combine-chunks`

**Review by kou — CHANGES_REQUESTED** [review]

File: c_glib/test/test-table.rb (hunk 1 above)
Comment (mrkn) [reply c_glib/test/test-table.rb]: I see. I'll rewrite it. Thanks.

File: c_glib/arrow-glib/table.cpp (hunk 2 above)
Comment (mrkn) [reply c_glib/arrow-glib/table.cpp]: I see.

**Review by kou — APPROVED** [review]
+1
Thanks!

**Pull request closed without merging**